### PR TITLE
Add INTELLIJ_SETUP.md

### PR DESCRIPTION
**Pull request by drcrallen**
This is an example of some of the stuff in our internal onboarding guide.

### CODE REVIEW

File: INTELLIJ_SETUP.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
+# IntelliJ Setup
+This document contains some examples and instructions on how to get IntelliJ setup to run local debuging and test setups of Druid.
```
Comment (gianm) [inline INTELLIJ_SETUP.md]: debugging (sp)

File: INTELLIJ_SETUP.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+## Code Style
+The Code Style is available in XML format at [druid_intellij_formatting.xml](druid_intellij_formatting.xml) and can be [imported into IntelliJ](https://www.jetbrains.com/help/idea/2017.1/copying-code-style-settings.html).
+
+## MySQL
```
Comment (gianm) [inline INTELLIJ_SETUP.md]: How about calling this "Metadata store" and mentioning that PostgreSQL and Derby could also work? (I usually use Derby in local testing fwiw)

Comment (drcrallen) [inline INTELLIJ_SETUP.md]: I don't have a means of testing that :) if you have better info to add here, that would be cool but probably outside of this PR.

Comment (gianm) [inline INTELLIJ_SETUP.md]: Could you at least add that PostgreSQL and Derby could be made to work too, although this document will only be describing how to get going with MySQL?

I just don't want people to get the wrong idea and think that MySQL is required.

Comment (gianm) [inline INTELLIJ_SETUP.md]: i.e., getting this running with other stores is left as an exercise for the reader.

Comment (drcrallen) [inline INTELLIJ_SETUP.md]: Sure, that's reasonable!

Comment (drcrallen) [inline INTELLIJ_SETUP.md]: @gianm fixed

**Pull request merged**